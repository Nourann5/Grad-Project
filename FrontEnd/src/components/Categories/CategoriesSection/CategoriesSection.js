import React from 'react'
import { Col, Container, Row } from 'react-bootstrap'
import styles from './CategoriesSection.module.css'

import PageCategoryCard from 'components/Global/Elements/PageCategoryCard/PageCategoryCard'
import PageContentHeader from 'components/Global/Elements/PageContentHeader/PageContentHeader'

import {ReactComponent as AmericanIcon} from 'assets/icons/USA.svg'
import {ReactComponent as AzharIcon} from 'assets/icons/azhar.svg'
import {ReactComponent as LocalIcon} from 'assets/icons/egypt.svg'

function CategoriesSection() {
    const cardData = [
        {
            cardText:'American',
            cardIcon:<AmericanIcon/>,
            cardLink:'/all-exams'
        },
        {
            cardText:'Local',
            cardIcon:<LocalIcon/>,
            cardLink:'/all-exams'
        },
        {
            cardText:'Alazhar',
            cardIcon:<AzharIcon/>,
            cardLink:'/all-exams'
        }
    ]
    return (
        <section className={styles["show-categories"]}>
            <Container fluid='lg'>
                <PageContentHeader content='Choose Your Category'/>
                <div className={styles["show-categories__cats"]}>
                    <Row>
                        {
                            cardData && cardData.map((card,index)=>(
                                <Col md='4' key={card.cardText +index}>
                                    <PageCategoryCard cardInfo={card} />
                                </Col>
                                
                            ))
                        }
                    </Row>
                </div>
            </Container>
        </section>
    )
}

export default CategoriesSection