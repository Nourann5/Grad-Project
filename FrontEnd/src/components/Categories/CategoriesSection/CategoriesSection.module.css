.show-categories{
    min-height: calc(100vh - 174px);
}

.show-categories__cats{
    display:flex;
    align-items: center;
    justify-content: space-around;
}