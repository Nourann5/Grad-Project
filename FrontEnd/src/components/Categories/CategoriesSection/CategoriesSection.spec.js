import {render , screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import CategoriesSection from './CategoriesSection'

describe('Test the Categories', () => { 
    it('heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <CategoriesSection/>
            </BrowserRouter>
        )
        expect(tree).toMatchSnapshot()
    })
 })