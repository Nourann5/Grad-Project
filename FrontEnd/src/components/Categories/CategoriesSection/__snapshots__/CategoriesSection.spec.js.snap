// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Categories heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="show-categories"
      >
        <div
          class="container-lg"
        >
          <h2
            class="page-content__header"
          >
            Choose Your Category
          </h2>
          <div
            class="show-categories__cats"
          >
            <div
              class="row"
            >
              <div
                class="col-md-4"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
              <div
                class="col-md-4"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
              <div
                class="col-md-4"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="show-categories"
    >
      <div
        class="container-lg"
      >
        <h2
          class="page-content__header"
        >
          Choose Your Category
        </h2>
        <div
          class="show-categories__cats"
        >
          <div
            class="row"
          >
            <div
              class="col-md-4"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
            <div
              class="col-md-4"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
            <div
              class="col-md-4"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
